### PR TITLE
update script to generate raptor report to login with managed ID

### PR DESCRIPTION
**Pull request by fey101**
Part of https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_workitems/edit/24118/
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoftgraph/microsoft-graph-devx-api/pull/2096)

### CODE REVIEW

File: pipelines/snippets.yml [hunk 1]
```diff
@@ -13,156 +13,21 @@ schedules:
...
- repositories:
-   - repository: microsoft-graph-docs
-     type: github
-     endpoint: microsoftgraph (22)
-     name: microsoftgraph/microsoft-graph-docs
-     ref: main
-
 pool:
-  vmImage: 'ubuntu-latest'
+  name: 1es-ubuntu-latest
```
Comment (fey101) [inline pipelines/snippets.yml:17]:
```suggestion
  name: 'ubuntu-latest'
```

File: pipelines/snippets.yml [hunk 2]
```diff
@@ -13,156 +13,21 @@ schedules:
...
- repositories:
-   - repository: microsoft-graph-docs
-     type: github
-     endpoint: microsoftgraph (22)
-     name: microsoftgraph/microsoft-graph-docs
-     ref: main
-
 pool:
-  vmImage: 'ubuntu-latest'
+  name: 'ubuntu-latest'
```
Comment (fey101) [inline pipelines/snippets.yml:17]:
```suggestion
  vmImage: 'ubuntu-latest'
```

File: scripts/getWorkloadOwner.ps1 [hunk 3]
```diff
@@ -27,16 +27,32 @@ function Get-AppSettings ()
...
+        az login --identity -u $env:RAPTOR_CONFIGMANAGEDIDENTITY_ID
+        # az login
+        az account show
+        Write-Host "Login successful. Fetching AppSettings from Azure App Config."
+    }
+    catch {
+        Write-Host "Failed to login using Managed Identity."
+    }
+    try {
+        az appconfig kv export --endpoint $env:RAPTOR_CONFIGENDPOINT --auth-mode login --key * --label $settingsLabel --destination file --path $appSettingsPath --format json --yes
```
Comment (fey101) [inline scripts/getWorkloadOwner.ps1:48]:
```suggestion
        az appconfig kv export --endpoint $env:RAPTOR_CONFIGENDPOINT --auth-mode login --label $settingsLabel --destination file --path $appSettingsPath --format json --yes
```

File: scripts/getWorkloadOwner.ps1 [hunk 4]
```diff
@@ -27,16 +27,38 @@ function Get-AppSettings ()
...
+        # az login
+        Write-Host "Login successful. Fetching AppSettings from Azure App Config."
+    }
+    catch {
+        Write-Host "Failed to login using Managed Identity."
+    }
+    try {
+        az appconfig kv export --endpoint $env:RAPTOR_CONFIGENDPOINT --auth-mode login --label $settingsLabel --destination file --path $appSettingsPath --format json --yes
+        $appSettings = Get-Content $AppSettingsPath -Raw | ConvertFrom-Json
+        Write-Host "AppSettings fetched successfully. Tenant ID value:$($appSettings.TenantID)"
```
Comment (andrueastman) [inline scripts/getWorkloadOwner.ps1:56]:
```suggestion
```
Do you need the value logged?

Comment (fey101) [reply scripts/getWorkloadOwner.ps1:56]: Just to confirm the az appconfig kv export is working as expected. Part of debugging. I don't see any problem in logging tenant ID do you?

Comment (andrueastman) [reply scripts/getWorkloadOwner.ps1:56]: I'm not sure if we really need the data in the logs. When debugging one could always add their own log statements yes?
Furthermore, line 63 in the scripts already validates the value is present...

File: scripts/categorizeErrors.ps1 [hunk 5]
```diff
@@ -89,6 +92,7 @@ foreach ($trxFilePath in $files){
 
         # assign result to appropriate category
         $specificError = $result.specificError
+        write-host "Categorizing $specificError"
```
Comment (fey101) [inline scripts/categorizeErrors.ps1:95]:
```suggestion
```

**Review by andrueastman — APPROVED** [review]

**Pull request merged**